SPM package resolution fails due to dynamic FileManager check in Package.swift

## Problem

When adding tree-sitter-matlab as a Swift Package Manager dependency, package resolution fails with:

```
exhausted attempts to resolve the dependencies graph, with the following dependencies unresolved:
* 'tree-sitter-matlab' from https://github.com/acristoffers/tree-sitter-matlab
```

## Cause

Two issues in `Package.swift`:

### 1. Dynamic FileManager check fails in SPM sandbox

```swift
var sources = ["src/parser.c"]
if FileManager.default.fileExists(atPath: "src/scanner.c") {
    sources.append("src/scanner.c")
}
```

`FileManager.fileExists()` returns `false` in SPM's sandboxed environment, so `scanner.c` is never included. This causes a **build failure** with undefined symbols:

```
Undefined symbols for architecture arm64:
  "_tree_sitter_matlab_external_scanner_create", referenced from:
      _tree_sitter_matlab.language in parser.c.o
  ...
```

### 2. Test target dependency reference

```swift
dependencies: [
    "SwiftTreeSitter",  // ❌ bare string doesn't work
    "TreeSitterMatlab",
]
```

SPM requires explicit product references when the package identity (`swift-tree-sitter`) differs from the product name (`SwiftTreeSitter`). This causes a **resolution failure**:

```
unknown dependency 'SwiftTreeSitter' in target 'TreeSitterMatlabTests'; 
valid dependencies are: 'swift-tree-sitter'
```

## Fix

```swift
// 1. Static source list
sources: ["src/parser.c", "src/scanner.c"],

// 2. Explicit product reference
dependencies: [
    .product(name: "SwiftTreeSitter", package: "swift-tree-sitter"),
    "TreeSitterMatlab",
]
```

## Status

Fixed on fork: https://github.com/kkilchrist/tree-sitter-matlab — PR incoming.

## Environment

- macOS 26.2 (25C56)
- Xcode 26.2 (17C48)

--- a/Package.swift
+++ b/Package.swift
@@ -1,12 +1,6 @@
 // swift-tools-version:5.3
 
-import Foundation
 import PackageDescription
-
-var sources = ["src/parser.c"]
-if FileManager.default.fileExists(atPath: "src/scanner.c") {
-    sources.append("src/scanner.c")
-}
 
 let package = Package(
     name: "TreeSitterMatlab",
@@ -21,7 +15,7 @@ let package = Package(
             name: "TreeSitterMatlab",
             dependencies: [],
             path: ".",
-            sources: sources,
+            sources: ["src/parser.c", "src/scanner.c"],
             resources: [
                 .copy("queries")
             ],
@@ -31,7 +25,7 @@ let package = Package(
         .testTarget(
             name: "TreeSitterMatlabTests",
             dependencies: [
-                "SwiftTreeSitter",
+                .product(name: "SwiftTreeSitter", package: "swift-tree-sitter"),
                 "TreeSitterMatlab",
             ],
             path: "bindings/swift/TreeSitterMatlabTests"

